orttraining/orttraining/test/training_api/core/training_api_tests.cc

```diff
@@ -162,8 +179,19 @@ void TestLRSchduler(const std::string& test_file_name, float initial_lr, int64_t
   CheckpointState state;
   auto checkpoint_to_load_path = MODEL_FOLDER "checkpoint.ckpt";
   ORT_ENFORCE(LoadCheckpoint(checkpoint_to_load_path, state).IsOK());
-  auto model = std::make_unique<Module>(model_uri, state.module_checkpoint_state.named_parameters);
-  auto optim = std::make_shared<Optimizer>(optim_uri, state.module_checkpoint_state.named_parameters);
+
+  onnxruntime::SessionOptions session_option;
+  std::unique_ptr<Environment> env;
+  ORT_THROW_IF_ERROR(Environment::Create(nullptr, env));
+  auto module_sess = std::make_unique<onnxruntime::InferenceSession>(session_option, *env);
+  auto optim_sess = std::make_unique<onnxruntime::InferenceSession>(session_option, *env);
+  ORT_THROW_IF_ERROR(module_sess->Load(model_uri));
+  ORT_THROW_IF_ERROR(module_sess->Initialize());
+  ORT_THROW_IF_ERROR(optim_sess->Load(optim_uri));
+  ORT_THROW_IF_ERROR(optim_sess->Initialize());
+
```

Comment:
Can you add in a CUDA device test here?

Reply:
We indeed need GPU tests for all components. I will add a new PR along with other left-over issues. Don't want to block you guys for further iterations based on the e2e.